apps/AppWithWearable/lib/pages/home/settings.dart

```diff
@@ -317,6 +323,10 @@ class SettingsBottomSheet extends StatelessWidget {
           return;
         }
         saveSettings();
+         if (!await keysAreValid()) {
```

Comment:
- Can we execute this inside `saveSettings `.
- Also, if useFriendKeys, not trigger the keys validation.
- And trigger the test prompt only if there's a modification in the new key that is being saved.
- Finally, if I have the keyboard expanded, I can't see the toast notification, until the keyboard is hidden.

Reply:
Thanks @josancamon19, appreciate the thorough review. I updated it and implemented your feedback, I think this new way is more elegant.